tests/flamegraph.rs

```diff
@@ -1,34 +1,65 @@
 extern crate inferno;
 
-use inferno::flamegraph::{self, BackgroundColor, Options, Palette};
+use inferno::flamegraph::{self, BackgroundColor, Direction, Options, Palette};
+use log::Level;
 use std::fs::{self, File};
-use std::io::{BufRead, BufReader, Cursor};
+use std::io::{self, BufRead, BufReader, Cursor};
 use std::path::PathBuf;
 use std::str::FromStr;
 
-fn test_flamegraph(input_file: &str, expected_result_file: &str, options: Options) {
+fn test_flamegraph(
+    input_file: &str,
+    expected_result_file: &str,
+    options: Options,
+) -> quick_xml::Result<()> {
     let r = File::open(input_file).unwrap();
     let expected_len = fs::metadata(expected_result_file).unwrap().len() as usize;
     let mut result = Cursor::new(Vec::with_capacity(expected_len));
-    flamegraph::from_reader(options, r, &mut result).unwrap();
-    let mut expected = BufReader::new(File::open(expected_result_file).unwrap());
+    let return_value = flamegraph::from_reader(options, r, &mut result);
+    let expected = BufReader::new(File::open(expected_result_file).unwrap());
+    result.set_position(0);
```

Comment:
I don't think you explicitly have to set the position to 0?

Reply:
We're using `result` as both a writer and reader. It gets passed into `flamegraph::from_reader` to write the result into, which advances the cursor. Then when we're comparing we need to reset the cursor so we start reading from the beginning.